tracer/src/Datadog.Trace/Configuration/supported-configurations.json

```diff
@@ -0,0 +1,1588 @@
+      ],
+      "product": "Iast"
+    },
+    "DD_IAST_DB_ROWS_TO_TAINT": {
+      "version": [
+        "A"
+      ],
+      "product": "Iast"
+    },
+    "DD_IAST_DEDUPLICATION_ENABLED": {
+      "version": [
+        "A"
+      ],
+      "product": "Iast"
+    },
+    "DD_IAST_ENABLED": {
+      "version": [
+        "A"
+      ],
+      "product": "Iast"
+    },
+    "DD_IAST_MAX_CONCURRENT_REQUESTS": {
+      "version": [
+        "A"
+      ],
+      "product": "Iast"
+    },
+    "DD_IAST_MAX_RANGE_COUNT": {
+      "version": [
+        "A"
+      ],
+      "product": "Iast"
+    },
+    "DD_IAST_REDACTION_ENABLED": {
+      "version": [
+        "A"
+      ],
+      "product": "Iast"
+    },
+    "DD_IAST_REDACTION_NAME_PATTERN": {
+      "version": [
+        "A"
+      ],
+      "product": "Iast"
+    },
+    "DD_IAST_REDACTION_VALUE_PATTERN": {
+      "version": [
+        "A"
+      ],
+      "product": "Iast"
+    },
+    "DD_IAST_REGEXP_TIMEOUT": {
+      "version": [
+        "A"
+      ],
+      "product": "Iast"
+    },
+    "DD_IAST_REQUEST_SAMPLING": {
+      "version": [
+        "A"
+      ],
+      "product": "Iast"
+    },
+    "DD_IAST_TELEMETRY_VERBOSITY": {
+      "version": [
+        "A"
+      ],
+      "product": "Iast"
+    },
+    "DD_IAST_TRUNCATION_MAX_VALUE_LENGTH": {
+      "version": [
+        "A"
+      ],
+      "product": "Iast"
+    },
+    "DD_IAST_VULNERABILITIES_PER_REQUEST": {
+      "version": [
+        "A"
+      ],
+      "product": "Iast"
+    },
+    "DD_IAST_WEAK_CIPHER_ALGORITHMS": {
+      "version": [
+        "A"
+      ],
+      "product": "Iast"
+    },
+    "DD_IAST_WEAK_HASH_ALGORITHMS": {
+      "version": [
+        "A"
+      ],
+      "product": "Iast"
+    },
+    "DD_INJECTION_ENABLED": {
+      "version": [
+        "A"
+      ],
+      "product": "Profiler"
+    },
+    "DD_INSTRUMENTATION_TELEMETRY_AGENTLESS_ENABLED": {
+      "version": [
+        "A"
+      ],
+      "product": "Telemetry"
+    },
+    "DD_INSTRUMENTATION_TELEMETRY_AGENT_PROXY_ENABLED": {
+      "version": [
+        "A"
+      ],
+      "product": "Telemetry"
+    },
+    "DD_INSTRUMENTATION_TELEMETRY_ENABLED": {
+      "version": [
+        "A"
+      ],
+      "product": "Telemetry"
+    },
+    "DD_INSTRUMENTATION_TELEMETRY_URL": {
+      "version": [
+        "A"
+      ],
+      "product": "Telemetry"
+    },
+    "DD_INTERNAL_RCM_POLL_INTERVAL": {
+      "version": [
+        "A"
+      ],
+      "product": "Rcm"
+    },
+    "DD_INTERNAL_TELEMETRY_DEBUG_ENABLED": {
+      "version": [
+        "A"
+      ],
+      "product": "Telemetry"
+    },
+    "DD_INTERNAL_WAIT_FOR_DEBUGGER_ATTACH": {
+      "version": [
+        "A"
+      ]
+    },
+    "DD_INTERNAL_WAIT_FOR_NATIVE_DEBUGGER_ATTACH": {
+      "version": [
+        "A"
+      ]
+    },
+    "DD_LOGS_DIRECT_SUBMISSION_AZURE_FUNCTIONS_HOST_ENABLED": {
+      "version": [
+        "A"
+      ],
+      "product": "DirectLogSubmission"
```

Comment:
Can we update the `product` name later? Or does this need to be decided right now and be set in stone? The reason I ask is because I think the various `DD_LOGS_DIRECT_SUBMISSION_*` flags and `DD_LOGS_OTEL_ENABLED` could all be categorized as a `"Logs"` product, but I would rather have that as a separate conversation and not block your work

Reply:
Yes, the product is something I added for the dotnet tracer to reflect already existing inner classes but it's not set in stone. But that's a good point, how to refactor autogenerated elements? From experience, you can ctrl+R,R on the inner class in the ConfigurationKeys.g.cs which will refactor everywhere in the solution, then change find and replace product name in the supported-configurations.json, so that the newly generated file is the same as what you just refactored. Not sure this is clear lol, I think I'll add something in the docs about this